public.tf

```diff
@@ -0,0 +1,74 @@
+module "public_label" {
+  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.2.2"
+  namespace  = "${var.namespace}"
+  name       = "${var.availability_zone}"
+  attributes = ["public"]
+  stage      = "${var.stage}"
+  delimiter  = "${var.delimiter}"
+  tags       = "${var.tags}"
+}
+
+resource "aws_subnet" "public" {
+  count             = "${var.type == "public" ? length(var.names) : 0}"
+  vpc_id            = "${var.vpc_id}"
+  availability_zone = "${var.availability_zone}"
+  cidr_block        = "${cidrsubnet(var.cidr_block, ceil(log(var.max_subnets_number, 2)), count.index)}"
+
+  tags = {
+    "Name"      = "${module.public_label.id}${var.delimiter}${element(var.names, count.index)}"
+    "Stage"     = "${module.public_label.stage}"
+    "Namespace" = "${module.public_label.namespace}"
+  }
+}
+
+resource "aws_route_table" "public" {
+  count  = "${var.type == "public" ? length(var.names) : 0}"
+  vpc_id = "${var.vpc_id}"
+
+  tags = {
+    "Name"      = "${module.public_label.id}${var.delimiter}${element(var.names, count.index)}"
+    "Stage"     = "${module.public_label.stage}"
+    "Namespace" = "${module.public_label.namespace}"
+  }
+}
+
+resource "aws_route" "public" {
+  count                  = "${var.type == "public" ? length(var.names) : 0}"
+  route_table_id         = "${element(aws_route_table.public.*.id, count.index)}"
+  gateway_id             = "${var.igw_id}"
+  destination_cidr_block = "0.0.0.0/0"
+}
+
+resource "aws_route_table_association" "public" {
+  count          = "${var.type == "public" ? length(var.names) : 0}"
+  subnet_id      = "${element(aws_subnet.public.*.id, count.index)}"
+  route_table_id = "${element(aws_route_table.public.*.id, count.index)}"
+}
+
+resource "aws_network_acl" "public" {
+  count      = "${var.type == "public" && signum(length(var.private_network_acl_id)) == 0 ? 1 : 0}"
```

Comment:
`signum ()` doesn't necessary here, you can use just `length(var.private_network_acl_id) > 0` it will boolean result too

Reply:
This change will not affect anything.
It sounds more like personal preferences.
Besides, you may find several use cases with `signum(length))` in the other modules.
https://github.com/cloudposse/terraform-aws-dynamic-subnets/blob/master/private.tf#L51

I don't think, this change should be a strict requirement.

Reply:
yea, it's just cosmetic :) , but it's good practice when you use smaller count of steps for getting same result

Reply:
Let's do cosmetic as a separate PR.